### ISSUE TITLE
Crafting timer.

### ISSUE THREAD

**Issue by NeviNovat** [CleverRaven/Cataclysm-DDA#46155]
# Is your feature request related to a problem? Please describe.
I'm frustrated when I start a long craft ( >4 hours ) and I want to only craft for less then that for any reason ( to prevent focus drain, maybe I want to take advantage of daylight for other thing, to respect an in-game schedule/in-game sleep schedule ) I have to pay attention to stop the crafting progress before I craft for more time then I want. This means that on performant platforms you pretty much have a "quick time event" where you have to be ready to stop your crafting the moment you start, and on less performant platforms you just sit and stare at a clock for the time to pass just paying attention because you can't leave and do something else or your character will craft till something else stops them ( hunger , thirst , rest or enemy sightings ).

# Describe the solution you'd like
Add a pop up box for crafts that take longer then 1 hour that lets you chose the amount of time you wish to spend on crafting.
Like the one you get before you try to sleep, or the one that you get when wait for time to pass. ( but maybe not make it require a watch for this option to be available )
**Maybe** with the option to craft till you start feeling hungry/thirsty/tiered/unfocused or craft till dawn/weather change (like how its implemented in the wait function).

# Describe alternatives you've considered
There is already an alternative in game ... you get a watch and set its alarm ( with the wait option ) to whatever you want and then start crafting. This uses more steps, is tedious for some, can be **forgotten** about or not **known** by players.
Combined with the fact that now certain crafting recipes take longer if your character is not proficient, ( a feature that **I like** since now you need to work on a big project for multiple days if you don't have a background in it, but ) the need to stop crafting and **resume** then stop again is even greater.
We already have a similar function for when you sleep, that asks you when you want to set your alarm clock why not for crafting when it takes more then a certain amount of time?

# Additional context
I would like for the ability to stop even when i don't have a watch, based on the progress of the recipe.

When you **don't** posses an item that tells time:

> Crafting this item will require more then 1 hour. How long would you like to spend crafting?
> 1 Till you finish crafting
> 2 Till you reach 25% of the crafting progress
> 3 Till you reach 50% of the crafting progress
> 4 Till you reach 75% of the crafting progress
> d Till dawn
> n Till noon
> k Till dusk

When you **do** posses an item that tells time:

> Crafting this item will require more then 1 hour. How long would you like to spend crafting?
> 1 Till you finish crafting
> 2 30 minutes
> 3 1 hour
> 4 2 hours
> 5 4 hours
> d Till dawn
> n Till noon
> k Till dusk

Or with even more fine control, like this:

> Crafting this item will require more then 45 minutes. How long would you like to spend crafting?
> 1 Till you finish crafting
> 2 15 minutes
> 3 30 minutes
> 4 1 hour
> 5 2 hours
> 6 3 hours
> 7 4 hours
> 8 6 hours
> 9 8 hours
> d Till dawn
> n Till noon
> k Till dusk

**Comment by Svankensen** [comment]
I'd prefer not to add extra key presses. So maybe have it ask IF you press something once you've started crafting?

**Comment by anothersimulacrum** [comment]
Either the threshold for popup should be greater or it should be a button pressed before starting the craft. (Or both)

**Comment by Svankensen** [comment]
> Either the threshold for popup should be greater or it should be a button pressed before starting the craft. (Or both)

You know, we could just add an alarm functionality to the clocks, and make it work like any other interruption. so before you craft you set up an alarm for X time in the future.

**Comment by anothersimulacrum** [comment]
That already exists, under `|`.

**Comment by Svankensen** [comment]
Isn't that only for waiting? I meant you set up an alarm and it interrupts whatever you are doing when it rings.

**Comment by anothersimulacrum** [comment]
![image](https://user-images.githubusercontent.com/42699974/102705393-37688600-423c-11eb-8106-766af9d216ea.png)
![image](https://user-images.githubusercontent.com/42699974/102705407-5109cd80-423c-11eb-9373-0206db042dee.png)

**Comment by Svankensen** [comment]
Duh, my bad, forgot I don't have an alarm clock in this character.

**Comment by anothersimulacrum** [comment]
It is mentioned in the alternatives section of the issue, I think that having this more integrated into the crafting menu would be beneficial.

**Comment by NeviNovat** [comment]
> 
> 
> It is mentioned in the alternatives section of the issue, I think that having this more integrated into the crafting menu would be beneficial.

Yes this is what I am asking for, and the ability for you to craft till 50% for example for when you don't have a way of telling time. Because right now if you craft without a clock I think you have no way to know **when** to stop if you want.

**Comment by Raikiri** [comment]
I don't see myself using a feature if it was implemented and I'd not like to have an extra keypress every time I craft. Alternatively, I don't mind at all if alarms received a better/more intuitive UI to be used for more stuff like reading.
